### PR TITLE
Calendar and map

### PR DESCRIPTION
**Pull request by mamrooz**
Added
Events veiw. (Calendar and Map)
GetEvents Controller
GetCategories Controller
mapAndCalendar.js and .css

### CODE REVIEW

File: app/SymfonyRequirements.php [hunk 1]
```diff
@@ -643,15 +643,15 @@ class_exists('Locale'),
             'Install and enable the <strong>intl</strong> extension (used for validators).'
         );
 
-        if (class_exists('Collator')) {
+        if (extension_loaded('intl')) {
```
Comment (smajazayeri) [inline app/SymfonyRequirements.php]: @mamrooz: Why did you remove this line? It sounds a critical change!!!

File: app/SymfonyRequirements.php [hunk 2]
```diff
@@ -643,15 +643,15 @@ class_exists('Locale'),
...
+            // in some WAMP server installations, new Collator() returns null
             $this->addRecommendation(
                 null !== new Collator('fr_FR'),
                 'intl extension should be correctly configured',
                 'The intl extension does not behave properly. This problem is typical on PHP 5.3.X x64 WIN builds.'
             );
-        }
 
-        if (class_exists('Locale')) {
+            // check for compatible ICU versions (only done when you have the intl extension)
```
Comment (smajazayeri) [inline app/SymfonyRequirements.php]: The same as above.

File: app/config/config.yml [hunk 3]
```diff
@@ -70,9 +70,9 @@ doctrine:
...
-    host:      "%mailer_host%"
-    username:  "%mailer_user%"
-    password:  "%mailer_password%"
-    spool:     { type: memory }
+#swiftmailer:
+#    transport: "%mailer_transport%"
+#    host:      "%mailer_host%"
+#    username:  "%mailer_user%"
+#    password:  "%mailer_password%"
+#    spool:     { type: memory }
```
Comment (smajazayeri) [inline app/config/config.yml]: Please, revert this change. The configs should be uncommented.

File: app/SymfonyRequirements.php (hunk 1 above)
Comment (smajazayeri) [inline app/SymfonyRequirements.php]: @mamrooz: Mine is also 2.6.9. can you please revert these changes on your machine and see whether it works fine. If not, we have to explore more about them.

File: app/config/config.yml~ [hunk 4]
```diff
@@ -0,0 +1,78 @@
+imports:
```
Comment (smajazayeri) [inline app/config/config.yml~]: you can delete this file.

File: app/config/parameters.yml~ [hunk 5]
```diff
@@ -0,0 +1,10 @@
+# This file is auto-generated during the composer install
+parameters:
+    database_driver: pdo_mysql
+    database_host: 127.0.0.1
+    database_port: null
+    database_name: eventyab_eventyab
+    database_user: eventyab_admin
+    database_password: Mohammad12345
+    locale: en
+    secret: ThisTokenIsNotSoSecretChangeIt
```
Comment (smajazayeri) [inline app/config/parameters.yml~]: The same as above.

File: composer.json [hunk 6]
```diff
@@ -21,7 +21,8 @@
         "sensio/distribution-bundle": "~3.0,>=3.0.12",
         "sensio/framework-extra-bundle": "~3.0,>=3.0.2",
         "incenteev/composer-parameter-handler": "~2.0",
-        "sensio/generator-bundle": "^2.5"
+        "sensio/generator-bundle": "^2.5",
+        "swiftmailer/swiftmailer": "@stable"
```
Comment (smajazayeri) [inline composer.json]: Interesting. You added the swiftmailer???? We're using it from the beginning!!!

Comment (smajazayeri) [inline composer.json]: Got it. We're using "symfony/swiftmailer-bundle". Please, revert the change in composer.json file. We don't need it.

File: src/Eventyab/HomeBundle/Controller/GetCategoriesController.php~ [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<?php
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Controller/GetCategoriesController.php~]: Please, delete this file.

File: src/Eventyab/HomeBundle/Resources/config/routing.yml~ [hunk 8]
```diff
@@ -0,0 +1,39 @@
+eventyab_home_getEvents:
+    path: /getEvents
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Resources/config/routing.yml~]: Please delete this file.

File: src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php~ [hunk 9]
```diff
@@ -0,0 +1,140 @@
+<?php
+use Symfony\Component\Finder\Finder;
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php~]: Please delete this file

File: src/Eventyab/HomeBundle/Resources/views/Default/index.html.php~ [hunk 10]
```diff
@@ -0,0 +1,450 @@
+<?php use Eventyab\HomeBundle\Constants\HomeConstants; ?>
+<!DOCTYPE html>
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Resources/views/Default/index.html.php~]: Delete

File: web/js/mapAndCalendar.js [hunk 11]
```diff
@@ -0,0 +1,888 @@
+var weekday = new Array(7);
+weekday[0] = "Sun";
+weekday[1] = "Mon";
```
Comment (smajazayeri) [inline web/js/mapAndCalendar.js]: Since I couldn't comment on ".mapAndCalendar.js.swp", I comment here. What's that binary file? Can you delete it?

File: web/js/mapAndCalendar.js~ [hunk 12]
```diff
@@ -0,0 +1,888 @@
+var weekday = new Array(7);
```
Comment (smajazayeri) [inline web/js/mapAndCalendar.js~]: Please delete this file.

File: src/Eventyab/HomeBundle/Controller/GetCategoriesController.php [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+    {
+	    $em = $this->getDoctrine()->getManager();
+	    $connection = $em->getConnection();
+	    $results = $connection->query("select name from category")->fetchAll();
+	    $res = array();
+	    foreach ($results as $r)
+		    array_push($res, $r['name']);
+	    return new Response(json_encode($res) );
+    }
+}
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Controller/GetCategoriesController.php]: Can you delete this file? I mean isn't it better to use our default controller? What's your opinion?

File: src/Eventyab/HomeBundle/Controller/GetEventsController.php~ [hunk 14]
```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Eventyab\HomeBundle\Controller;
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Controller/GetEventsController.php~]: Please, delete this file.

File: src/Eventyab/HomeBundle/Resources/config/routing.yml [hunk 15]
```diff
@@ -1,3 +1,11 @@
+eventyab_home_getEvents:
+    path: /getEvents
+    defaults: { _controller: EventyabHomeBundle:GetEvents:index }
+
+eventyab_home_getCategories:
+    path: /getCategories
+    defaults: { _controller: EventyabHomeBundle:GetCategories:index }
+
 eventyab_home_type:
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Resources/config/routing.yml]: To make it consistent, this file should contain routings belong to different bundles. Can you add these routings to here: https://github.com/mfivebusiness/Eventyab/blob/master/src/Eventyab/HomeBundle/Resources/config/routing.yml

File: src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+use Symfony\Component\Finder\Finder;
+$router = $this->get('router');
+$categoriesJSString =  $view['actions']->render(
+    new \Symfony\Component\HttpKernel\Controller\ControllerReference('EventyabHomeBundle:GetCategories:index')
+);
+$categories = json_decode($categoriesJSString);
+?>
+
+<link rel="stylesheet" type="text/css"
+	href="<?php echo $view['assets']->getUrl('css/mapAndCalendar.css') ?>";>
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php]: As it's not inside the head block, does this styling work???

File: src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php [hunk 17]
```diff
@@ -0,0 +1,140 @@
...
+					style="position: absolute; right: 10px; bottom: 10px; cursor: pointer;"
+					onClick="closeEventDetailsWindow()">Close</div>
+			</div>
+
+		</div>
+
+	</div>
+</div>
+
+<script type="text/javascript">
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php]: Can you please move the 2 scripts called at the top of this file, to here on line 116?

File: web/css/mapAndCalendar.css [hunk 18]
```diff
@@ -0,0 +1,431 @@
...
+	min-width: 500px;
+	padding: 0px;
+	margin: 0px;
+}
+
+#listOfEvents {
+	min-width: 200px;
+}
+
+.controls {
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: remove this class as we have it in mine.css.

File: web/css/mapAndCalendar.css [hunk 19]
```diff
@@ -0,0 +1,431 @@
+@CHARSET "ISO-8859-1";
+#map-canvas {
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: remove this class as we have it in mine.css.
What we have in mine.css is in line 296 which works fine although it has just one attribute. Please, give it a try and if your map visualization failed, ignore this comment of mine.

File: web/js/mapAndCalendar.js [hunk 20]
```diff
@@ -0,0 +1,888 @@
...
+
+	}
+
+	initializeMap = function() {
+		var markers = [];
+		map = new google.maps.Map(document.getElementById('map-canvas'),
+				mapOptions);
+
+		// Create the search box and link it to the UI element.
+		var input = /** @type {HTMLInputElement} */
```
Comment (smajazayeri) [inline web/js/mapAndCalendar.js]: you can delete this line as there is no element with this id (pac-input).

File: web/css/mapAndCalendar.css [hunk 21]
```diff
@@ -0,0 +1,431 @@
...
+	border: 1px solid transparent;
+	border-radius: 2px 0 0 2px;
+	box-sizing: border-box;
+	-moz-box-sizing: border-box;
+	height: 32px;
+	outline: none;
+	box-shadow: 0 2px 6px rgba(0, 0, 0, 0.3);
+}
+
+#pac-input {
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: delete it. it's not used anywhere in your code.

File: web/css/mapAndCalendar.css [hunk 22]
```diff
@@ -0,0 +1,431 @@
...
+	font-family: Roboto;
+	font-size: 15px;
+	font-weight: 300;
+	margin-left: 12px;
+	padding: 0 11px 0 13px;
+	text-overflow: ellipsis;
+	width: 400px;
+}
+
+#pac-input:focus {
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: the same as above.

File: web/css/mapAndCalendar.css [hunk 23]
```diff
@@ -0,0 +1,431 @@
...
+	border-color: #4d90fe;
+}
+
+.pac-container {
+	font-family: Roboto;
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: the same as above.

File: web/css/mapAndCalendar.css [hunk 24]
```diff
@@ -0,0 +1,431 @@
...
+}
+
+#type-selector {
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: the same as above.

File: web/css/mapAndCalendar.css [hunk 25]
```diff
@@ -0,0 +1,431 @@
...
+	color: #fff;
+	padding: 5px 11px 0px 11px;
+}
+
+#type-selector label {
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: the same as above.

File: web/css/mapAndCalendar.css [hunk 26]
```diff
@@ -0,0 +1,431 @@
...
+
+.FunEvent {
+	
+}
+
+.infoWindow {
+	width: 300px;
+}
+
+/* .myButton { */
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: delete these comments.

File: web/css/mapAndCalendar.css [hunk 27]
```diff
@@ -0,0 +1,431 @@
...
+	font-family: Roboto;
+	font-size: 15px;
+	font-weight: bold;
+	padding: 6px 24px;
+	text-decoration: none;
+	text-shadow: 0px 1px 0px #ffffff;
+}
+
+.myButton:hover {
+	background: -webkit-gradient(linear, left top, left bottom, color-stop(0.05, #f6f6f6
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: couldn't you use bootstrap classes for buttons to make our design consistent? http://www.w3schools.com/bootstrap/bootstrap_buttons.asp. If you have a specific reason, that's fine and ignore this comment of mine.

File: web/css/mapAndCalendar.css [hunk 28]
```diff
@@ -0,0 +1,431 @@
...
+
+.monthName {
+	font-size: small;
+}
+
+.year {
+	font-size: smaller;
+}
+
+.listOfEvents {
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: can you please remove the #lisOfEvents mentioned at the top of this file (line 10)?

File: web/css/mapAndCalendar.css [hunk 29]
```diff
@@ -0,0 +1,431 @@
...
+	width: 100%;
+	/* 	display: block; */
+	display: table-row;
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: I highly recommend you to use bootstrap classes as styling via display, width, height, ... might cause problems in tablets, cellphones or cross browsers.

File: web/css/mapAndCalendar.css [hunk 30]
```diff
@@ -0,0 +1,431 @@
...
+	white-space: nowrap;
+	overflow-y: scroll;
+	border: 0;
+	border-radius: 6px;
+	border-top: 0px;
+	z-index: 5;
+	text-align: center;
+	border: 1px solid;
+	max-height: 0;
+	white-space: nowrap;
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: you used it twice. remove the one at line 206.

File: web/css/mapAndCalendar.css [hunk 31]
```diff
@@ -0,0 +1,431 @@
...
+.hideListOfEvents {
+	max-height: 0;
+	transition: max-height .05s ease-out;
+}
+
+.dayDiv {
+	border-collapse: inherit;
+	border-spacing: 3px;
+	border-radius: 6px;
+/* 	display: inline-block; */
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: remove this comment.

File: web/css/mapAndCalendar.css [hunk 32]
```diff
@@ -0,0 +1,431 @@
...
+	margin-left: 5px;
+	margin-right: 5px;
+	border: 1px solid #dcdcdc;
+	width: 40px;
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: again, couldn't you use a bootstrap class instead of defining width, margin, ... here?

File: web/css/mapAndCalendar.css [hunk 33]
```diff
@@ -0,0 +1,431 @@
...
+
+.calendarDiv * {
+	color: Black;
+}
+
+.calendarDiv * :selection {
+	background: #a8d1ff;
+}
+
+h1, h2, h3, h4, h5 {
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: Please remove it as it influences on the previous usage of these elements.

File: web/css/mapAndCalendar.css [hunk 34]
```diff
@@ -0,0 +1,431 @@
...
+}
+
+h4 {
+	font-size: 18px;
+}
+
+h5 {
+	font-size: 14px;
+	color: #FF8400;
+}
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: the same as above.

File: web/css/mapAndCalendar.css [hunk 35]
```diff
@@ -0,0 +1,431 @@
...
+
+/* menu */
+
+.clearfix:after {
+    display:block;
+    clear:both;
+}
+ 
+/*----- Menu Outline -----*/
+.menu-wrap {
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: please use bootstrap classes. We SHOULDn't use width anymore.

File: web/css/mapAndCalendar.css [hunk 36]
```diff
@@ -0,0 +1,431 @@
...
+ 
+/*----- Bottom Level -----*/
+.menu li:hover .sub-menu {
+    z-index:1;
+    opacity:1;
+    display: block;
+}
+ 
+.sub-menu {
+    width:200%;
```
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: please remove it.

File: web/js/mapAndCalendar.js [hunk 37]
```diff
@@ -0,0 +1,888 @@
...
+		text += "</div>";
+		text += "<div style='width:300px;white-space: nowrap; overflow: hidden; text-overflow: ellipsis;'>";
+		text += this.address;
+		text += "</div>";
+		text += "</div>\n";
+		return text;
+	}
+
+	this.mathchesFilters = function(filters) {
+		// //console.log( this.starts.valueOf()/1000 + " > " +
```
Comment (smajazayeri) [inline web/js/mapAndCalendar.js]: please remove these comments.

File: web/js/mapAndCalendar.js [hunk 38]
```diff
@@ -0,0 +1,888 @@
...
+						&& this.eventDays[i].valueOf() >= filters.temporalFilterStart)
+					flag = true;
+			}
+			if (!flag)
+				return false;
+		} else {
+			if (filters.temporalFilterEnd != null
+					&& this.starts.valueOf() > filters.temporalFilterEnd)
+				return false;
+		//	//console.log("2");
```
Comment (smajazayeri) [inline web/js/mapAndCalendar.js]: remove.

File: web/js/mapAndCalendar.js [hunk 39]
```diff
@@ -0,0 +1,888 @@
...
+			if (filters.temporalFilterStart != null
+					&& this.ends.valueOf() < filters.temporalFilterStart)
+				return false;
+		}
+
+	//	//console.log("3");
```
Comment (smajazayeri) [inline web/js/mapAndCalendar.js]: the same.

File: web/js/mapAndCalendar.js [hunk 40]
```diff
@@ -0,0 +1,888 @@
...
+				// //console.log("'" + name + "' '" + filterString + "'");
+				if (name.indexOf(filterString) > -1)
+					flag = true;
+			}
+			if (!flag)
+				return false;
+			}else
+				return false;
+		}
+	//	//console.log("4");
```
Comment (smajazayeri) [inline web/js/mapAndCalendar.js]: the same.

File: web/js/mapAndCalendar.js [hunk 41]
```diff
@@ -0,0 +1,888 @@
...
+		 * //console.log("Matches Filter"); if (sH * 60 + sI != 0 || eH * 60 + eI !=
+		 * 24 * 60 - 1) { var startMin = this.starts.getHours() * 60 +
+		 * this.starts.getMinutes(); var endMin = this.ends.getHours() * 60 +
+		 * this.ends.getMinutes(); endMin = (endMin - startMin + (24 * 60)) %
+		 * (24 * 60); var startCriteria = (sH * 60 + sI + (24 * 60) - startMin) %
+		 * (24 * 60); var endCriteria = (eH * 60 + eI + (24 * 60) - startMin) %
+		 * (24 * 60); if (endMin < endCriteria && endMin < startCriteria) return
+		 * false; } if (categorialFilters.indexOf(this.category) == -1) return
+		 * false;
+		 */
```
Comment (smajazayeri) [inline web/js/mapAndCalendar.js]: please remove all comments and console.logs from this file.

File: src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php [hunk 42]
```diff
@@ -0,0 +1,140 @@
...
+?>
+
+<link rel="stylesheet" type="text/css"
+	href="<?php echo $view['assets']->getUrl('css/mapAndCalendar.css') ?>";>
+<script
+	src="<?php echo $view['assets']->getUrl('js/mapAndCalendar.js') ?>";></script>
+<script
+	src="https://maps.googleapis.com/maps/api/js?v=3.exp&signed_in=true&libraries=places"></script>
+<div
+	style="position: relative; min-width: 100%; height: 100%; min-height: 800px;">
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php]: Please, never ever use inline styling.

Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php]: Try to use bootstrap classes such as container>row>col-lg-??

File: src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php [hunk 43]
```diff
@@ -0,0 +1,140 @@
...
+	<div id="map-canvas"></div>
+	<input id="pac-input" class="controls" type="text"
+		placeholder="Navigate to"
+		style="position: absolute; top: 100px; left: 65px; opacity: .8">
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php]: the same as above.

File: src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php [hunk 44]
```diff
@@ -0,0 +1,140 @@
...
+								placeholder="Search By Event Name"
+								onkeyup="calMap.filterByName(this.value)"></a></li>
+					</ul></li>
+			</ul>
+		</nav>
+	</div>
+
+
+	<div
+		style="position: absolute; width: calc(100% -40px); top: 15px; left: 20px; right: 20px;">
```
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php]: please remove all inline stylings from this file.

**Comment by smajazayeri** [comment]
@mamrooz: Mamroz jan, please follow the comments and let me know if you disagree with any of them. Thanks.

File: src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php (hunk 16 above)
Comment (smajazayeri) [inline src/Eventyab/HomeBundle/Resources/views/Default/calendarAndMap.html.php]: @mamrooz: For example, now I need to know the answer of this comment. PLEASE, reply comments later.

File: web/css/mapAndCalendar.css (hunk 18 above)
Comment (smajazayeri) [inline web/css/mapAndCalendar.css]: @mamrooz: Agha, age man margin-top ro bokonam 10px, to moshkeli dari? Noktash ine ke address bar e mape man 6px balatar az baghieye elementame va niaz daram ke margin-top eh ro kam konam.

Comment (mamrooz) [inline web/css/mapAndCalendar.css]: Are OKe.
Momkene yekam mape man be ham berize.
Age didam be ham mirize man id e map-canvase khodamo avaz mikonam. Ke ba
ham overlap nadashte bashan.

Comment (mfivebusiness) [inline web/css/mapAndCalendar.css]: Haan. Eival. Pas moshkeli bud, begu lotfan.
